yoga/Yoga.cpp

```diff
@@ -2475,24 +2503,23 @@ static void YGJustifyMainAxis(
         break;
       case YGJustifySpaceBetween:
         if (collectedFlexItemsValues.itemsOnLine > 1) {
-          betweenMainDim =
+          betweenMainDim +=
               YGFloatMax(collectedFlexItemsValues.remainingFreeSpace, 0) /
               (collectedFlexItemsValues.itemsOnLine - 1);
-        } else {
-          betweenMainDim = 0;
         }
         break;
       case YGJustifySpaceEvenly:
         // Space is distributed evenly across all elements
-        betweenMainDim = collectedFlexItemsValues.remainingFreeSpace /
+        leadingMainDim = collectedFlexItemsValues.remainingFreeSpace /
             (collectedFlexItemsValues.itemsOnLine + 1);
-        leadingMainDim = betweenMainDim;
+        betweenMainDim += leadingMainDim;
         break;
       case YGJustifySpaceAround:
         // Space on the edges is half of the space between elements
-        betweenMainDim = collectedFlexItemsValues.remainingFreeSpace /
+        leadingMainDim = 0.5 *
+            collectedFlexItemsValues.remainingFreeSpace /
             collectedFlexItemsValues.itemsOnLine;
-        leadingMainDim = betweenMainDim / 2;
+        betweenMainDim += leadingMainDim * 2;
```

Comment:
could be a nit pick - this can also be represented as below

```
case YGJustifySpaceAround: {
        const float betweenMainDimWithoutGap = collectedFlexItemsValues.remainingFreeSpace /
            collectedFlexItemsValues.itemsOnLine;
        // Space on the edges is half of the space between elements
        leadingMainDim = betweenMainDimWithoutGap / 2;
        betweenMainDim += betweenMainDimWithoutGap;
        break;
      }
```

need to check if cost of multiplication > cost of defining a scope and a variable.

Reply:
Deja vu! I actually did it like this originally - then I thought this path is used a lot less than the flex start, and the extra variable could affect the entire function.

Reply:
Let me know what you think anyway. I’m not adverse to changing it

Reply:
As a thought, we could do,

```cpp
        leadingMainDim = collectedFlexItemsValues.remainingFreeSpace /
             collectedFlexItemsValues.itemsOnLine;
        betweenMainDim += leadingMainDim;
        leadingMainDim *= 0.5;
```

Reply:
yeah, this might be good as it prevents additional multiplication, but we can wait for someone with c++ performance experience to review it 😅